transport/src/main/java/io/netty/channel/pool/SimpleChannelPool.java

```diff
@@ -184,7 +184,7 @@ private void notifyHealthCheck(Future<Boolean> future, Channel ch, Promise<Chann
         assert ch.eventLoop().inEventLoop();
 
         if (future.isSuccess()) {
-            if (future.getNow() == Boolean.TRUE) {
+            if (future.getNow()) {
```

Comment:
@windie actually  I think this should stay like it was as getNow() really return a Boolean.